griptape/tasks/base_task.py

```diff
@@ -168,3 +174,15 @@ def full_context(self) -> dict[str, Any]:
             return structure_context
         else:
             return {}
+
+    def _add_to_structure(self, task: Optional[BaseTask]):
+        if task is not None:
+            if self.structure is not None and task.structure is not None and self.structure != task.structure:
+                raise ValueError("Task is already associated with a different structure.")
+            elif self.structure:
+                if task not in self.structure.tasks:
+                    self.structure.add_task(task)  # pyright: ignore reportArgumentType
```

Comment:
Why the pyright ignore?

Reply:
it says that `BaseTask | None` is not valid for `add_task`. even though theres a None check

Reply:
oh it passes now